src/core/object/controller/controller.ts

```diff
@@ -683,6 +711,41 @@ export default class Controller {
 		this.onModeChanged();
 	}
 
+	/**
+	 * Checks if the temp icon/mode is visible.
+	 */
+	private isTempIconVisible() {
+		return this.timeoutId !== undefined;
+	}
+
+	/**
+	 * Temporarily set the mode of the controller,
+	 * then returns to previous mode after 5 seconds.
+	 *
+	 * @param pastMode - past controller mode
+	 * @param newMode - new controller mode to be set
+	 *
+	 */
+	private setTempMode(
+		pastMode: ControllerModeStr,
+		newMode: ControllerModeStr,
+	) {
+		if (this.isTempIconVisible()) {
+			clearTimeout(this.timeoutId);
+			this.timeoutId = undefined;
+		} else {
+			this.tempMode = newMode;
+			this.mode = this.tempMode;
+		}
+		const TEMP_ICON_DISPLAY_DURATION = 5000;
+		this.setMode(newMode);
+		this.timeoutId = setTimeout(() => {
+			this.timeoutId = undefined;
+			this.setMode(pastMode);
+			this.tempMode = null;
+		}, TEMP_ICON_DISPLAY_DURATION);
+	}
```

Comment:
```suggestion
	private setTempMode(
		pastMode: ControllerModeStr,
		newMode: ControllerModeStr,
	) {
		if (this.isTempIconVisible()) {
			clearTimeout(this.timeoutId);
			this.timeoutId = undefined;
		}
		const TEMP_ICON_DISPLAY_DURATION = 5000;
		this.tempMode = newMode;
		this.onModeChanged();
		this.timeoutId = setTimeout(() => {
			this.timeoutId = undefined;
			this.tempMode = null;
			this.onModeChanged();
		}, TEMP_ICON_DISPLAY_DURATION);
	}
```
Untested, but we should be able to do something along these lines to completely avoid mutating this.mode and still get the changes shown.

Reply:
https://github.com/web-scrobbler/web-scrobbler/pull/4472/commits/80e606639a2728ec0d3f23dd729de37470c2868e